config/base/rbac/kustomization.yaml

```diff
@@ -9,3 +9,12 @@ resources:
 - role_binding.yaml
 - leader_election_role.yaml
 - leader_election_role_binding.yaml
```

Comment:
It seems accurate for me, but to ensure that the metrics is working as should be you might want use the same tests which are now scaffolded by kubebuilder, see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/test/e2e/e2e_test.go#L166-L235

Reply:
That sounds good! @kashifest can you try it?

Reply:
So, from the example above  https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/test/e2e/e2e_test.go#L166-L235

You can mainly do the tests by copying it and:
- Just remove https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/test/e2e/e2e_test.go#L180C1-L183C68
- Ensure that instead of project-v4 it uses the name of your project

Reply:
HI @kashifest 

I am looking at your project the way that it is missing:

```yaml
       - /manager
        args:
        - "--metrics-bind-address=:8443"
        - "--leader-elect"
        ports:
          - containerPort: 8443
            protocol: TCP
            name: https
```

I think that will fix the issue.

Reply:
Yes my local tests just passed with that. pushing it now

Reply:
HI @kashifest 

I can see that it is not passing in the tests. My bet here is that the NS where the manager is deployed is configured to allow admit only Pods which are using the Security Context.

So, you probably need to change the curl pod for something like

```
cmd = exec.Command("kubectl", "run", curlPod,
		"--image=curlimages/curl:7.87.0", "-n", namespace,
		"--restart=Never",
		"--overrides", `{
			"spec": {
				"containers": [{
					"name": "curl",
					"image": "curlimages/curl:7.87.0",
					"command": ["sh", "-c", "sleep 3600"],
					"securityContext": {
						"allowPrivilegeEscalation": false,
						"capabilities": {
							"drop": ["ALL"]
						},
						"runAsNonRoot": true,
						"runAsUser": 1000,
						"seccompProfile": {
							"type": "RuntimeDefault"
						}
					}
				}],
				"serviceAccountName": "THE MANAGER SA"
			}
		}`)
```

Reply:
Honestly I have no clue now why the test is failing, heres a screenshot of my local test where everything is happy 

![Uploading Screenshot 2024-12-13 at 14.56.34.png…]()

Reply:
It is missing the `command`. You can test it locally with the fixture test bu doing `make test-e2e`. It is fast and simple, no extra config needed.

Reply:
command was there and it was failing even then, I removed it for testing

Reply:
`make test-e2e` doesnt work as is, you need to build images also manually before it?

Reply:
Right, sorry. You need to do
```
IMG=quay.io/metal3-io/baremetal-operator:e2e make docker
make test-e2e
```

Reply:
ok, finally I was able to fix the metrics endpoint test, but it is now failing in upgrade. So new challenges 😄